dagster_publish_mdm/assets/tailscale/tailscale_devices.py

```diff
@@ -80,3 +81,70 @@ def tailscale_insert_and_update_devices(context: dg.AssetExecutionContext) -> tu
         f"Updated {updated_devices} and inserted {new_devices} devices into tailscale_device"
     )
     return updated_devices, new_devices
+
+
+@dg.asset(
+    group_name="tailscale_device_prunning_assets",
+    deps=["tailscale_device_snapshot"],
+    description="Scans devices with no activity for over 90 days",
+)
+def stale_tailscale_devices(
+    context: dg.AssetExecutionContext, tailscale_device_snapshot: dict
+) -> list:
+    """Scan for old tailscale devices"""
+
+    context.log.info("Scanning for stale devices...")
+
+    now = datetime.now(timezone.utc)
+    time_delta = now - timedelta(
+        minutes=30
+    )  # Currently set to 30 minutes to enable testing. will be set to days=90 in prod
```

Comment:
One option could be to default to the larger period and for devs to set an environment variable for local testing

```suggestion
    time_delta = now - timedelta(
        minutes=int(
            os.getenv(
                "TAILSCALE_DEVICE_STALE_MINUTES",
                default=129600,  # 60 minutes * 24 hours * 90 days
            )
        )
    )
```